service/history/visibilityQueueTaskExecutor.go

```diff
@@ -89,12 +89,27 @@ func (t *visibilityQueueTaskExecutor) Execute(
 ) ([]metrics.Tag, bool, error) {
 	task := executable.GetTask()
 	taskType := queues.GetVisibilityTaskTypeTagValue(task)
+	namespaceTag, replicationState := getNamespaceTagAndReplicationStateByID(
+		t.shard.GetNamespaceRegistry(),
+		task.GetNamespaceID(),
+	)
 	metricsTags := []metrics.Tag{
-		getNamespaceTagByID(t.shard.GetNamespaceRegistry(), task.GetNamespaceID()),
+		namespaceTag,
 		metrics.TaskTypeTag(taskType),
 		metrics.OperationTag(taskType), // for backward compatibility
 	}
 
+	if replicationState == enumspb.REPLICATION_STATE_HANDOVER {
```

Comment:
Any task need to be blocked in visibility queue?

Reply:
I commented in the code. close execution task will do a set workflow execution and will be blocked by the shard. 
From correctness perspective, I don't have to block any task here (even for timer/transfer), and can rely on shard to  block it. But then, upon retry task can still load mutable state which is allowed by shard context and fail upon write which waste resources. 

I can allow certain task types to pass like start execution etc. But I guess I want to keep things simple for now and block all tasks.